service/history/workflow/mutable_state_impl.go

```diff
@@ -275,8 +280,12 @@ func NewMutableState(
 		ExecutionStats:   &persistencespb.ExecutionStats{HistorySize: 0},
 	}
 	s.approximateSize += s.executionInfo.Size()
-	s.executionState = &persistencespb.WorkflowExecutionState{State: enumsspb.WORKFLOW_EXECUTION_STATE_CREATED,
-		Status: enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING}
+	s.executionState = &persistencespb.WorkflowExecutionState{
+		RunId: runID,
```

Comment:
what is the reason to add RunId to the ExecutionState and not to ExecutionInfo?

Reply:
Data structure fields wise, `RunID` has always been tracked inside `WorkflowExecutionState`; whereas `NamespaceID` and `WorkflowID` are tracked in `WorkflowExecutionInfo`

Reply:
could you kindly point me to the code? I do not see it inside the state, but I see some RunIds inside execution info.

Reply:
ah, found it.
different struct.